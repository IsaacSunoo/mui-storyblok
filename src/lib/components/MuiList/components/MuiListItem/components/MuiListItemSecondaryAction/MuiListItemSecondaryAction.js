import React, { createElement, lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import Storyblok from '../../../../../../utils/Storyblok';

const MuiIconButtonRedirect = lazy(() => import('../../../../../MuiIconButtonRedirect/MuiIconButtonRedirect'));
const MuiIconButtonHref = lazy(() => import('../../../../../MuiIconButtonHref/MuiIconButtonHref'));
const MuiIconButtonDownload = lazy(() => import('../../../../../MuiIconButtonDownload/MuiIconButtonDownload'));
const MuiIconButtonDialog = lazy(() => import('../../../../../MuiIconButtonDialog/MuiIconButtonDialog'));
const MuiTooltip = lazy(() => import('../../../../../MuiTooltip/MuiTooltip'));

/**
 * MuiListItemSecondaryAction is used in storyblok redirect to react routes
 */

export const MuiListItemSecondaryAction = ({
  content,
  rootClass,
}) => {
  const components = {
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiIconButtonDownload,
    MuiIconButtonDialog,
    MuiTooltip,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <ListItemSecondaryAction className={styles.root}>
      <Suspense fallback={<div />}>
        {content.map((item, index) => createElement(
          components[item.component],
          Object.assign(item, { key: index }),
        ))}
      </Suspense>
    </ListItemSecondaryAction>
  );
};

export default MuiListItemSecondaryAction;

MuiListItemSecondaryAction.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /** MuiButtonRedirect, MuiTooltip; Allowed maximum: 1 */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiListItemSecondaryAction.defaultProps = {
  rootClass: [],
};
