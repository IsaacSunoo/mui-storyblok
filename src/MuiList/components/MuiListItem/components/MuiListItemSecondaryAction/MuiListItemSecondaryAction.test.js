import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import MuiListItemSecondaryAction from './MuiListItemSecondaryAction';

function setup() {
  const props = {
    content: [{
      component: 'MuiIconButtonRedirect',
      buttonText: 'buttonText',
      redirectRoute: 'page-welcome',
    }],
  };
  const comp = shallow(<MuiListItemSecondaryAction {...props} />);
  return { comp, props };
}

describe('<MuiListItemSecondaryAction />', () => {
  it('renders MuiListItemSecondaryAction', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const mockstore = configureMockStore();
    const store = mockstore({});
    const tree = renderer.create((
      <Provider store={store}>
        <MemoryRouter>
          <MuiListItemSecondaryAction {...props} />
        </MemoryRouter>
      </Provider>
    ));
    expect(tree).toMatchSnapshot();
  });
});
