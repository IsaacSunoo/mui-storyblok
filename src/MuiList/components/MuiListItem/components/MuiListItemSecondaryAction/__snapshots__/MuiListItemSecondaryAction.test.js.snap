// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiListItemSecondaryAction /> snapshot 1`] = `
<div
  className="MuiListItemSecondaryAction-root makeStyles-root-3"
>
  <div />
</div>
`;
